datafusion/optimizer/src/scalar_subquery_to_join.rs

```diff
@@ -242,7 +245,7 @@ fn optimize_scalar(
     }
 
     // Only operate if one column is present and the other closed upon from outside scope
-    let subqry_alias = format!("__sq_{}", config.next_id());
+    let subqry_alias = alias.next("__scalar_sq");
```

Comment:
I debated passing an `Arc<AliasGenerator>` into each of the rules, this created a fair amount of churn, and it is easier for the rules to just assign a unique prefix. As an added bonus it highlights where the alias came from, which is nice I guess